### PR TITLE
Autoprune release notes included

### PR DESCRIPTION
**Pull request by mrbandrews**

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -1,6 +1,73 @@
...
+-reindex. Note also that any problem that would cause a user to reindex (e.g., 
+disk corruption) will cause a pruned node to redownload the entire blockchain. 
+Finally, note that when a pruned node reindexes, it will delete any blk???.dat 
+and rev???.dat files in the data directory prior to restarting the download.
+
+To enable block pruning on the command line:
+
+- `-prune=N`: where N is the number of MB to allot for raw block & undo data.
+
+Modified RPC calls:
```
Comment (jonasschnelli) [inline doc/release-notes.md]: Would it be a overkill to mention that the REST block call (`/rest/block/<blockhash>.<json|bin|hex>`) also could return a http 404 if a pruned block was requested?
(https://github.com/bitcoin/bitcoin/blob/e9af4e65b5afaa870f6cdcf2f7c2ee56c703bffa/src/rest.cpp#L200)

**Comment by jonasschnelli** [comment]
LGTM. ACK.
Didn't check the spelling / english.

**Comment by PRabahy** [comment]
Would it be worth it to include anything about switching wallets with pruning enabled? It is my understanding that if the blockfiles get pruned past where the wallet last updated that you will need to redownload the entire blockchain.

**Comment by sipa** [comment]
If you run with pruning enabled, you cannot have any wallet at all,
currently.

**Comment by PRabahy** [comment]
My bad. I thought #6057 had been merged.

**Comment by laanwj** [comment]
@PRabahy That's a common misconception, so it is good that it's even mentioned in this text that pruning currently doesn't interoperate with the wallet.

#6057 needs a lot of testing, but no matter what, it's too late for 0.11.

**Comment by laanwj** [comment]
ACK

**Comment by fanquake** [comment]
ACK

**Pull request merged**